src/redux/actions/planActions.js

```diff
@@ -0,0 +1,329 @@
+        'x-access-token': localStorage.getItem('token')
+      },
+      method: 'GET'
+    })
+      .then(response => response.json())
+      .then(json => {
+        if (json.success === false) dispatch(getPlansFailure());
+        else {
+          let plans = [];
+          json.result.forEach(t => {
+            if (plans[t.trip_id] === undefined) plans[t.trip_id] = {};
+            plans[t.trip_id][t.e_id] = t;
+          });
+          dispatch(getPlansSuccess(plans));
+        }
+      });
+  };
+}
+
+/**
+ * Performs an http POST addPlan request to server.
+ * Dispatches addPlan to indicate the beginning of an addPlan process.
+ * Dispatches addPlanFailure to indicate the end of a failed addPlan process.
+ * Dispatches addPlanSuccess to indicate the end of a successful addPlan process.
+ * If addPlan process succeeds, getPlanInfo is dispatched using the returned
+ * e_id of the newly added plan.
+ * @param plan object containing email, name, and dscript.
+ * @returns {function(*): Promise<Response | never>} dispatch results.
+ */
+export function addPlan(plan) {
+  return dispatch => {
+    dispatch(addPlanRequest()); // Add plan request process has begun...
+    return fetch('http://localhost:4201/plan/add', {
+      headers: {
+        'Content-Type': 'application/json',
+        'x-access-token': localStorage.getItem('token')
+      },
+      method: 'POST',
+      body: JSON.stringify(plan)
+    })
+      .then(response => response.json())
+      .then(json => {
+        if (json.success === false) dispatch(addPlanFailure());
+        else {
+          dispatch(addPlanSuccess());
+          dispatch(getPlanInfo(json.result)); // Fetch added plan
+        }
+      });
+  };
+}
+
+/**
+ * Performs an http PUT updatePlan request to server.
+ * Dispatches updatePlan to indicate the beginning of an updatePlan process.
+ * Dispatches updatePlanFailure to indicate the end of a failed updatePlan process.
+ * Dispatches updatePlanSuccess to indicate the end of a successful updatePlan process.
+ * If updatePlan process succeeds, the outdated plan is filtered out of the state list
+ * and getPlanInfo is dispatched using the e_id of the updated plan.
+ * @param plan object containing name, dscript, e_id.
+ * @returns {function(*): Promise<Response | never>} dispatch results.
+ */
+export function updatePlan(plan) {
+  return (dispatch, getState) => {
+    dispatch(updatePlanRequest()); // Update plan request process has begun...
+    return fetch('http://localhost:4201/plan/update', {
+      headers: {
+        'Content-Type': 'application/json',
+        'x-access-token': localStorage.getItem('token')
+      },
+      method: 'PUT',
+      body: JSON.stringify(plan)
+    })
+      .then(response => response.json())
+      .then(json => {
+        if (json.success === false) dispatch(updatePlanFailure());
+        else {
+          // Delete outdated plan, fetch updated plan
+          let plans = getState().masterReducer.plans.plans;
```

Comment:
Again, this should be a reducer right? What if this was wrapped into an action like removePlanSuccess(plan.trip_id, plan.e_id) which is sent to the reducer to do the heavy lifting?

Also, updatePlanSuccess isn't actually true until the fetch is complete.

Reply:
See my comment above. As for 
> Also, updatePlanSuccess isn't actually true until the fetch is complete.
I am open to a better Action name for this stage in the process. I agree that it doesn't seem to properly indicate what is happening at this point.